modules/fedimint-meta-server/src/db.rs

```diff
@@ -0,0 +1,96 @@
+use fedimint_core::encoding::{Decodable, Encodable};
+use fedimint_core::{impl_db_lookup, impl_db_record, PeerId};
+use fedimint_meta_common::{MetaConsensusValue, MetaKey, MetaValue};
+use serde::Serialize;
+use strum_macros::EnumIter;
+
+/// Namespaces DB keys for this module
+#[repr(u8)]
+#[derive(Clone, EnumIter, Debug)]
+pub enum DbKeyPrefix {
+    /// How we want to vote
+    ///
+    /// Private, not part of the consensus, but only local state.
+    Desired = 0x00,
+    /// Current consensuson
+    Consensus = 0x01,
+    /// Current submitted votes
+    Submissions = 0x02,
+}
+
+// TODO: Boilerplate-code
+impl std::fmt::Display for DbKeyPrefix {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{self:?}")
+    }
+}
+
+#[derive(Debug, Clone, Encodable, Decodable, Eq, PartialEq, Hash, Serialize)]
+pub struct MetaDesiredKey(pub MetaKey);
+
+#[derive(Debug, Encodable, Decodable)]
+pub struct MetaDesiredKeyPrefix;
+
+#[derive(Debug, Encodable, Decodable, Serialize, PartialEq, Eq)]
+pub struct MetaDesiredValue {
+    // Since AlephBft will merge and not re-submit the exact same item twice within one session,
+    // changing submitted item in sequence `a -> b -> a` will simply ignore the second `a`.
+    // To avoid this behavior, an otherwise meaningless `salt` field is used.
+    pub salt: u64,
```

Comment:
@joschisan something we might want to change since it's quite a foot gun

Reply:
Seems like some stuff already already relies on this behavior (including this PR), so ... not sure if we can change it now.

Does not seem like a big problem, but it is a footgun, so we should document it loudly in many places.